exasol_advanced_analytics_framework/lua/test/test_exasol_tools.lua

```diff
@@ -0,0 +1,64 @@
+local json = require('cjson')
+local luaunit = require("luaunit")
+local exasol_script_tools = require("exasol_script_tools")
+local mockagne = require("mockagne")
+
+test_exasol_script_tools = {
+}
+
+function test_exasol_script_tools.test_parse_arguments_query_correct_with_udf()
+    local functions_mock = mockagne.getMock()
+    exa_env = {
+        functions = functions_mock
+    }
+    local expected_table = {
+        temporary_output = {
+            bucketfs_location = {
+                connection_name = "bfs_conn",
+                directory = "directory"
+            },
+            schema_name = "temp_schema"
+        },
+        query_handler = {
+            class = {
+                name = "cls_name",
+                module = "package.module"
+            },
+            udf = {
+                schema = "UDF_SCHEMA",
+                name = "UDF_NAME"
+            },
+            parameters = "params"
+        },
+    }
+    local json_str = json.encode(expected_table)
+    local args = exasol_script_tools.parse_arguments(json_str, exa_env)
```

Comment:
Encoding and redecoding in a test is strictly speaking not very safe. I will let it pass in this particular case though, because JSON is simple enough. :-)

The clean way (but also more inconvenient way) would be to hardcode the JSON input string.

Reply:
yep, learned this a few minutes ago, I think, I can avoid with some refactoring